plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/TestMySqlCaseInsensitiveMapping.java

```diff
@@ -15,162 +15,60 @@
 
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
-import com.google.common.collect.ImmutableSet;
-import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.plugin.jdbc.BaseCaseInsensitiveMappingTest;
 import io.trino.testing.QueryRunner;
-import io.trino.testing.sql.TestTable;
+import io.trino.testing.sql.SqlExecutor;
 import org.testng.annotations.Test;
 
-import java.util.stream.Stream;
+import java.nio.file.Path;
 
-import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static io.trino.plugin.jdbc.mapping.RuleBasedIdentifierMappingUtils.createRuleBasedIdentifierMappingFile;
 import static io.trino.plugin.mysql.MySqlQueryRunner.createMySqlQueryRunner;
-import static io.trino.testing.assertions.Assert.assertEquals;
-import static java.lang.String.format;
-import static java.util.Locale.ENGLISH;
-import static org.assertj.core.api.Assertions.assertThat;
+import static java.util.Objects.requireNonNull;
 
 // With case-insensitive-name-matching enabled colliding schema/table names are considered as errors.
 // Some tests here create colliding names which can cause any other concurrent test to fail.
 @Test(singleThreaded = true)
 public class TestMySqlCaseInsensitiveMapping
-        extends AbstractTestQueryFramework
+        extends BaseCaseInsensitiveMappingTest
 {
+    private Path mappingFile;
     private TestingMySqlServer mySqlServer;
 
     @Override
     protected QueryRunner createQueryRunner()
             throws Exception
     {
+        mappingFile = createRuleBasedIdentifierMappingFile();
         mySqlServer = closeAfterClass(new TestingMySqlServer());
-        return createMySqlQueryRunner(mySqlServer, ImmutableMap.of(), ImmutableMap.of("case-insensitive-name-matching", "true"), ImmutableList.of());
+        return createMySqlQueryRunner(
+                mySqlServer,
+                ImmutableMap.of(),
+                ImmutableMap.<String, String>builder()
+                        .put("case-insensitive-name-matching", "true")
+                        .put("case-insensitive-name-matching.config-file", mappingFile.toFile().getAbsolutePath())
+                        .put("case-insensitive-name-matching.config-file.refresh-period", "1ms") // ~always refresh
```

Comment:
flaky?

Also is the refresh to the mapping synchronized on something? i.e. what happens if mapping gets updated during query planning?

Reply:
That is a very good question, so +1 from me

Reply:
I don't think it is flaky. This works because tests are serial. Mapping is set before a test and then it is used throughout the test.

Reply:
Thanks for explaining. But can it still not happen during normal operation? A refresh might be triggerred between calls to the various `toRemoteX` methods?

Reply:
That is correct. This a preexisting problem because we use something similar for file based access controls where access rule might change in the middle of the query. If we go further with this problem we might notice that HMS state can also change in the middle of the query analysis. Same goes to other connectors.

So far nobody complains about that this refreshing mechanism (for other places), notice that typically it is set to like a `10m` and I would expect that typically it is changed incrementally in small chunks. Here I would expect there is a single table or schema that is causing an issue and mapping is should not be ever changed. So I don't think this is an issue.